c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2724,6 +2738,20 @@ class DeviceCachingAllocator {
     cuda_events.clear();
   }
 
+  void remove_cudagraph_stream_uses(Block* block) {
+    // remove stream uses added during cudagraph capture
+    // (i.e., block->stream_uses - block->cudagraph_stream_uses)
+    stream_set streams(std::move(block->stream_uses));
+    AT_ASSERT(block->stream_uses.empty());
+    for (auto& stream : streams) {
+      if (block_to_cudagraph_stream_uses[block].find(stream) ==
+          block_to_cudagraph_stream_uses[block].end()) {
+        block->stream_uses.insert(stream);
+      }
+    }
+    block_to_cudagraph_stream_uses[block].clear();
```

Comment:
nit - might as well remove the block form the map instead of just clearing the stream set.